lib/vm/src/instance.rs

```diff
@@ -109,6 +105,45 @@ pub(crate) struct Instance {
     vmctx: VMContext,
 }
 
+/// TODO: figure out what to do with Clone here
+///       We can probably remove the Arc... just need to figure that out.
+/// TODO:
+#[derive(Debug)]
+pub struct ImportEnv {
+    /// TODO:
+    pub env: *mut std::ffi::c_void,
+    /// TODO:
+    pub clone: Option<fn(*mut std::ffi::c_void) -> *mut std::ffi::c_void>,
```

Comment:
Why is `clone` optional?

Reply:
That's a good question! I believe I've clarified this now by making all 3 states explicit with an `enum`!